### PR TITLE
Convert Descriptor ID to Indicator ID 

### PR DESCRIPTION
**Pull request by jeberl**
Summary
---------

We were in correctly storing the indicator ID as the banked content ID when fetching from ThreatUpdates.

1. Change simple serialization in `python-threatexchange` to store both indicator id and descriptor roll up in CSV output. (will need to bump pip version in later PR) This effects how csv's are stored in s3
2. Change the post apply function in the fetcher which reads csv rows and creates PDQMetadata objects to now store the indicator ID rather thann the descriptor ID
3. Change the s3_adapters to read the new column names from the CSV
4. Change the ThreatExchange writebacker to recognize that `id` now refers to indicator id not descriptor ID 

NOTE: After this lands terraform destroy is probably needed

Test Plan
---------
Retested after rebase

1. Make upload_docker and update lambdas
2. delete all data in s3
3. refetch ThreatExchange data in s3 and confirm the csv columns are correct
4. Watch as index is rebuilt
5. Upload image using HMA UI
6. Watch match, copy ID, and using ID tool confirm that the ID is the correct indicator ID:
Screen Shot 2021-05-06 at 9 27 41 AM

### CODE REVIEW

**Review by BarrettOlson — COMMENTED** [review]
This introduces a breaking change to python-threatexchange, @jeberl and I discussed possible alternatives and tradeoffs involved to this. The outcome of that discussion was that he is going to explore implementing a non-breaking serialization to work in HMA. 

Putting the PR back in draft mode for now.

File: python-threatexchange/threatexchange/descriptor.py
Comment (jeberl) [inline python-threatexchange/threatexchange/descriptor.py:180]: removing

File: hasher-matcher-actioner/hmalib/lambdas/fetcher.py
Comment (BarrettOlson) [inline hasher-matcher-actioner/hmalib/lambdas/fetcher.py:363]: rm before merge.

File: hasher-matcher-actioner/hmalib/lambdas/fetcher.py
Comment (BarrettOlson) [inline hasher-matcher-actioner/hmalib/lambdas/fetcher.py:259]: I missed this in #553 but having the constructor args for `indicator_type` and `indicator` be swapped between `HMASerialization` and `CliIndicatorSerialization` messed with my head. Maybe worth a comment.

File: hasher-matcher-actioner/hmalib/lambdas/fetcher.py
Comment (BarrettOlson) [inline hasher-matcher-actioner/hmalib/lambdas/fetcher.py:331]: I think we still want this to be 1...

**Review by BarrettOlson — APPROVED** [review]
Overall, look good. approving to unblock. A few nit in line to please address before merge as well as two others: 
- This will break `make dev_upload_test_signal` can you just comment it out in the makefile if you don't want to go through the trouble of changing it? 
- can you delete?https://github.com/facebook/ThreatExchange/blob/9224eb7a00c61ffcc489bcccd8f0a3a3d11caf9d/hasher-matcher-actioner/hmalib/lambdas/pdq/pdq_indexer.py#L25 (as it is unused and this change make it inaccurate)  

Thanks!

File: hasher-matcher-actioner/hmalib/lambdas/fetcher.py
Comment (jeberl) [reply hasher-matcher-actioner/hmalib/lambdas/fetcher.py:259]: Ah. Didnt think of that either. A comment would be useful but don't think this is the right place. Probably in the `HMASerialization` class

**Pull request merged**
